File: okhttputil/src/main/java/com/ashlikun/okhttputils/http/IOUtils.java
package com.ashlikun.okhttputils.http;

import android.text.TextUtils;

import java.io.Closeable;
import java.io.File;
import java.io.Flushable;
import java.io.IOException;

/**
 * 作者　　: 李坤
 * 创建时间: 2018/5/10 0010　下午 4:19
 * 邮箱　　：496546144@qq.com
 * <p>
 * 功能介绍：
 */
public class IOUtils {
    public static void closeQuietly(Closeable closeable) {
        if (closeable == null) {
            return;
        }
        try {
            closeable.close();
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    public static void flushQuietly(Flushable flushable) {
        if (flushable == null) {
            return;
        }
        try {
            flushable.flush();
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    public static boolean createFolder(String folderPath) {
        if (!TextUtils.isEmpty(folderPath)) {
            File folder = new File(folderPath);
            return createFolder(folder);
        }
        return false;
    }

    public static boolean createFolder(File targetFolder) {
        if (targetFolder.exists()) {
            if (targetFolder.isDirectory()) {
                return true;
            }
            targetFolder.delete();
        }
        return targetFolder.mkdirs();
    }

    public static boolean delFileOrFolder(String path) {
        return !TextUtils.isEmpty(path) && delFileOrFolder(new File(path));
    }


    public static boolean delFileOrFolder(File file) {
        if (file == null || !file.exists()) {
            // do nothing
        } else if (file.isFile()) {
            file.delete();
        } else if (file.isDirectory()) {
            File[] files = file.listFiles();
            if (files != null) {
                for (File sonFile : files) {
                    delFileOrFolder(sonFile);
                }
            }
            file.delete();
        }
        return true;
    }

    public static boolean createFile(File targetFile) {
        if (targetFile.exists()) {
            if (targetFile.isFile()) {
                return true;
            }
            delFileOrFolder(targetFile);
        }
        try {
            return targetFile.createNewFile();
        } catch (IOException e) {
            return false;
        }
    }


    public static boolean createNewFile(String filePath) {
        if (!TextUtils.isEmpty(filePath)) {
            File file = new File(filePath);
            return createNewFile(file);
        }
        return false;
    }

    public static boolean createNewFile(File targetFile) {
        if (targetFile.exists() && !targetFile.isFile()) {
            delFileOrFolder(targetFile);
        }
        if (!targetFile.exists()) {
            try {
                return targetFile.createNewFile();
            } catch (IOException e) {
                return false;
            }
        }
        return true;
    }
}

File: okhttputil/src/main/java/com/ashlikun/okhttputils/http/IOUtils.kt
package com.ashlikun.okhttputils.http

import android.text.TextUtils
import java.io.Closeable
import java.io.File
import java.io.Flushable
import java.io.IOException
import java.lang.Exception

/**
 * @author　　: 李坤
 * 创建时间: 2021.12.17 9:30
 * 邮箱　　：496546144@qq.com
 *
 *
 * 功能介绍：io操作的相关工具
 */
object IOUtils {
    fun closeQuietly(closeable: Closeable?) {
        try {
            closeable?.close()
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    fun flushQuietly(flushable: Flushable?) {
        try {
            flushable?.flush()
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    fun createFolder(folderPath: String): Boolean {
        return folderPath.isNotEmpty() && createFolder(File(folderPath))
    }

    fun createFolder(targetFolder: File): Boolean {
        if (targetFolder.exists()) {
            if (targetFolder.isDirectory) {
                return true
            }
            targetFolder.delete()
        }
        return targetFolder.mkdirs()
    }

    fun delFileOrFolder(path: String): Boolean {
        return path.isNotEmpty() && delFileOrFolder(File(path))
    }

    fun delFileOrFolder(file: File): Boolean {
        if (!file.exists()) {
        } else if (file.isFile) {
            file.delete()
        } else if (file.isDirectory) {
            val files = file.listFiles()
            if (files != null) {
                for (sonFile in files) {
                    delFileOrFolder(sonFile)
                }
            }
            file.delete()
        }
        return true
    }

    fun createFile(targetFile: File): Boolean {
        if (targetFile.exists()) {
            if (targetFile.isFile) {
                return true
            }
            delFileOrFolder(targetFile)
        }
        return try {
            targetFile.createNewFile()
        } catch (e: IOException) {
            false
        }
    }

    fun createNewFile(filePath: String): Boolean {
        return filePath.isNotEmpty() && createNewFile(File(filePath))
    }

    fun createNewFile(targetFile: File): Boolean {
        if (targetFile.exists() && !targetFile.isFile) {
            delFileOrFolder(targetFile)
        }
        return if (!targetFile.exists()) {
            try {
                targetFile.createNewFile()
            } catch (e: IOException) {
                false
            }
        } else true
    }
}